File: backend/src/main/java/com/meuprojeto/tmdb/TmdbController.java
package com.meuprojeto.tmdb;

import com.meuprojeto.catalog.ContentType;
import com.meuprojeto.tmdb.dto.TmdbImportResult;
import com.meuprojeto.tmdb.dto.TmdbSearchResult;
import lombok.RequiredArgsConstructor;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/tmdb")
@RequiredArgsConstructor
public class TmdbController {

    private final TmdbService tmdbService;

    @GetMapping("/status")
    public Map<String, Boolean> status() {
        return Map.of("configurado", tmdbService.isConfigured());
    }

    @GetMapping("/search")
    public List<TmdbSearchResult> buscar(@RequestParam String query, @RequestParam ContentType tipo) {
        return tmdbService.buscar(query, tipo);
    }

    @PostMapping("/import")
    @PreAuthorize("hasRole('ADMIN')")
    public TmdbImportResult importar(@RequestParam Long tmdbId, @RequestParam ContentType tipo) {
        return tmdbService.importar(tmdbId, tipo);
    }
}

File: backend/src/tmdb/tmdb.controller.ts
import { Controller, Get, Post, Query, UseGuards } from '@nestjs/common';
import { ApiTags } from '@nestjs/swagger';
import { Role } from '@prisma/client';
import { TmdbService } from './tmdb.service';
import { TmdbSearchQueryDto, TmdbImportQueryDto } from './dto/tmdb-query.dto';
import { Roles } from '../common/decorators/roles.decorator';
import { RolesGuard } from '../common/guards/roles.guard';

@ApiTags('tmdb')
@Controller('api/tmdb')
export class TmdbController {
  constructor(private readonly tmdbService: TmdbService) {}

  @Get('status')
  status() {
    return { configurado: this.tmdbService.isConfigured() };
  }

  @Get('search')
  buscar(@Query() query: TmdbSearchQueryDto) {
    return this.tmdbService.buscar(query.query, query.tipo);
  }

  @Post('import')
  @UseGuards(RolesGuard)
  @Roles(Role.ADMIN)
  importar(@Query() query: TmdbImportQueryDto) {
    return this.tmdbService.importar(query.tmdbId, query.tipo);
  }
}
